### PR TITLE
Initial pass of fetching data from gRPC backend and click to update

### PR DESCRIPTION
**Pull request by kwasi**
## Implementation

* Adds RPCs to fetch Input and Output executions for the target artifact
* Adds RPCs to fetch Input and Output artifacts for the input and output executions
* Pass down real Artifact and Execution objects to LineageCardRow and pass them up when an artifact is clicked. 
* Partial implementation of  #150 and #152

## Demo

* https://screencast.googleplex.com/cast/NDgxMjQzNDIyNzMzMTA3MnwwZjVjMDM2My00MQ
* https://kubeflow-ui-07.endpoints.kwasinti-kubeflow-dev.cloud.goog/_/metadata/#/artifact_types/kubeflow.org/alpha/data_set/artifacts/3

## Notes

* This is a pretty barebones PR providing just enough to get the RPC calls hooked up. 
* All RPCs are executed serially.
* There is no empty state
* There is no spinner when loading data
* Has not been tested with a large data set with multiple executions or deep lineage history

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/metadata/179)

### CODE REVIEW

**Comment by kwasi** [comment]
Updated https://kubeflow-ui-07.endpoints.kwasinti-kubeflow-dev.cloud.goog/ to use the latest version of this PR.

File: frontend/src/pages/LineageView.tsx [hunk 1]
```diff
@@ -115,29 +115,90 @@ class LineageView extends React.Component<LineageViewProps, LineageViewState> {
...
+      await this.metadataStoreService.getEventsByArtifactIDs(getEventsByArtifactIDsRequest);
+
+    const events = getEventsByArtifactIDsResponse.getEventsList();
+
+    const outputExecutionIds =
+      events.filter(isOutputEvent).map((event) => (event.getExecutionId()));
+    const inputExecutionIds =
+      events.filter(isInputEvent).map((event) => (event.getExecutionId()));
+
+    const outputExecutions = await this.getExecutions(outputExecutionIds);
```
Comment (avdaredevil) [inline frontend/src/pages/LineageView.tsx]: Let's parallelize this:
```suggestion
    const [outputExecutions, inputExecutions] = await Promise.all([
        this.getExecutions(outputExecutionIds),
        this.getExecutions(inputExecutionIds),
    ])
```

File: frontend/src/pages/LineageView.tsx [hunk 2]
```diff
@@ -115,29 +115,90 @@ class LineageView extends React.Component<LineageViewProps, LineageViewState> {
...
+        .filter(isOutputEvent)
+        .map((event) => event.getArtifactId());
+    const outputArtifacts = await this.getArtifacts(outputExecutionOutputArtifactIds);
+
+    this.setState({
+      inputArtifacts: inputArtifacts,
+      inputExecutions: outputExecutions,
+      outputExecutions: inputExecutions,
+      outputArtifacts: outputArtifacts,
+    });
```
Comment (avdaredevil) [inline frontend/src/pages/LineageView.tsx]: Condense the entire object as so:
```suggestion
    this.setState({
        inputArtifacts, inputExecutions, outputExecutions, outputArtifacts
    });
```

Comment (kwasi) [reply frontend/src/pages/LineageView.tsx]: Done

File: frontend/src/lib/Utils.tsx [hunk 3]
```diff
@@ -195,3 +202,31 @@ export function getExpandedRow(expandedRows: Map<number, Row[]>, columns: Column
...
+}
+
+function getExecutionName(execution: Execution): string {
+  return String(getResourceProperty(execution, ExecutionProperties.NAME))
+}
+
+export function getResourceName(resource: LineageResource): string {
+  if (resource instanceof Artifact) {
+    return getArtifactName(resource);
+  } else {
```
[the remark this answers is not in the document]
Comment (kwasi) [reply frontend/src/lib/Utils.tsx]: Done

File: frontend/src/pages/LineageView.tsx (hunk 2 above)
Comment (kwasi) [reply frontend/src/pages/LineageView.tsx]: Done.

File: frontend/src/pages/LineageView.tsx (hunk 1 above)
Comment (kwasi) [reply frontend/src/pages/LineageView.tsx]: Done.

File: frontend/src/pages/LineageView.tsx [hunk 4]
```diff
@@ -115,29 +115,90 @@ class LineageView extends React.Component<LineageViewProps, LineageViewState> {
...
+    const getEventsByArtifactIDsRequest = new GetEventsByArtifactIDsRequest();
+    getEventsByArtifactIDsRequest.addArtifactIds(id);
+
+    const getEventsByArtifactIDsResponse =
+      await this.metadataStoreService.getEventsByArtifactIDs(getEventsByArtifactIDsRequest);
+
+    const events = getEventsByArtifactIDsResponse.getEventsList();
+
+    const outputExecutionIds =
+      events.filter(isOutputEvent).map((event) => (event.getExecutionId()));
```
[the remark this answers is not in the document]
Comment (kwasi) [reply frontend/src/pages/LineageView.tsx]: Done.

File: frontend/src/pages/LineageView.tsx [hunk 5]
```diff
@@ -115,29 +115,90 @@ class LineageView extends React.Component<LineageViewProps, LineageViewState> {
...
+    const inputExecutionIds =
+      events.filter(isInputEvent).map((event) => (event.getExecutionId()));
```
[the remark this answers is not in the document]
Comment (kwasi) [reply frontend/src/pages/LineageView.tsx]: Done.

**Comment by avdaredevil** [comment]
/lgtm
/approve

Kwasi this looks great, thanks!

Quick Note: instead of merging master, try rebasing next time (in order to not introduce remerges and extra commits into the parent tree)

**Pull request merged**